subprojects/core/src/main/java/org/gradle/api/Task.java

```diff
@@ -359,6 +359,19 @@ public String determineName(Task c) {
     Task doFirst(Closure action);
 
     /**
+     * <p>Adds the given closure to the beginning of this task's action list. The closure is passed this task as a
+     * parameter when executed.</p>
+     *
+     * @param actionName An arbitrary string that is used for logging.
+     * @param action The action closure to execute.
+     * @return This task.
+     *
+     * @since 4.2
+     */
+    @Incubating
+    Task doFirst(String actionName, Closure action);
```

Comment:
Sorry, I should have included my reasoning...

We don't need separate `Closure` taking methods because Groovy can use the `Action` methods directly because `Action` has a single method and Groovy can coerce closures into whatever type.  If you delete the closure taking methods here, your tests will still work.